src/providers/registry.ts

```diff
@@ -449,7 +449,11 @@ export const providerMap: ProviderFactory[] = [
     ) => {
       // Load script module
       const scriptPath = providerPath.split(':')[1];
-      return new ScriptCompletionProvider(scriptPath, providerOptions);
+      // Resolve relative paths against context.basePath or current working directory
+      // If using a cloud config, always use process.cwd() instead of context.basePath
+      const isCloudConfig = providerOptions.config?.isCloudConfig === true;
```

Comment:
Can you abstract this into a helper? Could even start with `const scriptPath = providerPath.startsWith('file://')`